src/pages/home/report/ReportActionItem.js

```diff
@@ -19,30 +24,152 @@ const propTypes = {
+        return !_.isEqual(this.props, nextProps) || !_.isEqual(this.state, nextState);
+    }
+
+    /**
+     * Save the location of a native press event.
+     *
+     * @param {Object} nativeEvent
+     */
+    capturePressLocation(nativeEvent) {
+        this.popoverAnchorX = nativeEvent.screenX;
+        this.popoverAnchorY = nativeEvent.screenY;
+    }
+
+    /**
+     * Show the ReportActionContextMenu modal popover.
+     *
+     * @param {Object} [event] - A press event.
+     */
+    showModal(event) {
+        const nativeEvent = event.nativeEvent || {};
+        this.capturePressLocation(nativeEvent);
+
+        // If the popover will display off-screen, scroll the ReportActionsView FlatList down to this item first.
+        if (this.popoverAnchorY - this.popoverHeight < 0) {
+            this.popoverAnchorY += (this.popoverHeight + 20);
+            this.props.scrollToThis(this.popoverHeight + 20);
+        }
+
+        if (!this.props.isSmallScreenWidth) {
+            // On large screens, only display the ReportActionContextMenu on RightClick, not LongPress.
+            if (isRightClick(nativeEvent)) {
+                this.setState({isModalVisible: true});
+            }
+        } else {
+            this.setState({isModalVisible: true});
+        }
+    }
+
+    /**
+     * Hide the ReportActionContextMenu modal popover.
+     */
+    hideModal() {
+        this.setState({isModalVisible: false});
+    }
+
+    /**
+     * Measure the size of the ReportActionContextMenu popover.
+     *
+     * @param {Object} nativeEvent
+     */
+    measurePopover({nativeEvent}) {
+        this.popoverWidth = nativeEvent.layout.width;
+        this.popoverHeight = nativeEvent.layout.height;
+    }
+
+    render() {
+        const {getContainerStyle, getModalStyleOverride} = getReportActionItemStyles();
+        return (
+            <PressableWithSecondaryInteraction onSecondaryInteraction={this.showModal}>
+                <Hoverable>
+                    {hovered => (
+                        <View>
+                            <View style={getContainerStyle(hovered)}>
+                                {!this.props.displayAsGroup
+                                    ? <ReportActionItemSingle action={this.props.action} />
+                                    : <ReportActionItemGrouped action={this.props.action} />}
+                            </View>
+                            <View style={styles.miniReportActionContextMenuWrapperStyle}>
+                                <ReportActionContextMenu
+                                    reportID={this.props.reportID}
+                                    reportActionID={this.props.action.sequenceNumber}
+                                    isVisible={hovered && !this.state.isModalVisible}
+                                    isMini
+                                />
+                            </View>
+                            <Modal
```

Comment:
I missed this the first time around but I think we are maybe violating the cross platform philosophy of the app here since there are some platforms that will never show a modal and some that will never show the hoverable menu yet we are showing both here (which makes it somewhat difficult to understand in what contexts they will show). 

Probably how we should structure this is:

`index.js` - show both (actually not sure why we need to show both hover _and_ right click style menu, but that's a different conversation)
`index.native.js` - only include the modal version (In this case, I think it does not need to be wrapped in a `Hoverable`)

I understand we are sort of solving this with the `Hoverable` in a way, but I'd argue that there's a lot of extra noise now and it's hard to tell what's being used where.

Reply:
> I understand we are sort of solving this with the Hoverable in a way, but I'd argue that there's a lot of extra noise now and it's hard to tell what's being used where

Fair point. I've updated this to separate out the web and native implementations. That way, the native code won't all this extra stuff in the render function, which will hopefully improve performance on the native platforms too.